service/history/engine/interface.go

```diff
@@ -67,6 +67,7 @@ type (
 		RespondCrossClusterTasksCompleted(ctx context.Context, targetCluster string, responses []*types.CrossClusterTaskResponse) error
 		QueryWorkflow(ctx context.Context, request *types.HistoryQueryWorkflowRequest) (*types.HistoryQueryWorkflowResponse, error)
 		ReapplyEvents(ctx context.Context, domainUUID string, workflowID string, runID string, events []*types.HistoryEvent) error
+		CountDLQMessages(ctx context.Context, forceFetch bool) (map[string]int64, error)
```

Comment:
Question: Do you think there's any chance of this needing to return more than just a count value? Would a response struct be worthwhile in the event that there is potentially further stuff to return?

Reply:
I have added more general structure in proto, so that additional fields could be added into key later if needed.

This interface is internal, so we can refactor it later if needed. For now I wanted to keep it simple.